importlib_metadata/__init__.py

```diff
@@ -199,16 +199,31 @@ def __iter__(self):
         warnings.warn(msg, DeprecationWarning)
         return iter((self.name, self))
 
-    def __reduce__(self):
-        return (
-            self.__class__,
-            (self.name, self.value, self.group),
-        )
-
     def matches(self, **params):
         attrs = (getattr(self, param) for param in params)
         return all(map(operator.eq, params.values(), attrs))
 
+    def _key(self):
+        return tuple(getattr(self, key) for key in 'name value group'.split())
+
+    def __lt__(self, other):
```

Comment:
I would recommend the total_ordering decorator from the stdlib

Reply:
I'd like to deprecate ordering support, as I'm not sure it's valuable, so total_ordering would go against that goal. I'll defer adding total_ordering until it has value.